packages/lib/gpcircuits/src/util.ts

```diff
@@ -11,3 +11,57 @@ export function artifactPaths(
     vkeyPath: path.join(root, cd.name + "-vkey.json")
   };
 }
+
+/**
+ * Splits an array `arr` into chunks of size
+ * `n` in order. If `arr.length` is not a multiple of `n`,
+ * then the last chunk will be of length `arr.length % n`.
+ */
+export function toChunks<A>(arr: A[], n: number): A[][] {
+  return arr.reduce(
+    (chunks: A[][], a: A) => {
+      const lastChunkIndex = chunks.length - 1;
+      const lastChunk = chunks[lastChunkIndex];
+      return lastChunk.length < n // If the last chunk is incomplete
+        ? chunks.slice(0, lastChunkIndex).concat([lastChunk.concat([a])]) // Append the current value to it.
+        : chunks.concat([[a]]); // Else start a new chunk with the current value in it.
+    },
+    [[]]
+  );
+}
+
+/**
+ * Applies a Promise-valued function to array elements
+ * sequentially. Necessary to avoid OOM for particularly
+ * heavy computations.
+ */
+export async function seqPromise<A, B>(
+  f: (a: A) => Promise<B>,
+  arr: A[]
+): Promise<B[]> {
+  let outputArray: B[] = [];
+
+  for (const a of arr) {
```

Comment:
Glad this worked out since I'd have been confused otherwise.
I'd argue that this function is now so simple as to not need a function (callers could just do the for loop) but it doesn't hurt.